src/FuelClient.cc

```diff
@@ -1425,8 +1443,7 @@ Result FuelClient::DownloadModel(const common::URI &_modelUrl,
   }
 
   _path = gz::common::joinPaths(this->Config().CacheLocation(),
-      id.Server().Url().Path().Str(), id.Owner(), "models", id.Name(),
-      id.VersionStr());
+      id.UniqueName(), id.VersionStr());
```

Comment:
I noticed that `UniqueName` uses `Server().Url().Str()`, while the code here previously had `Server().Url().Path().Str()`. Is this intentional?

Reply:
This is because previously when we had a web URL, we weren't setting the authority and treating the whole thing as a path, when really the domain name is the authority in this case

```
# Before (authority = false)
https://fuel.gazebosim.com/user/model
  Authority: ""
  Path: "fuel.gazebosim.com/user/model"

https://foo@localhost:8080/user/model -> 
  Authority: ""
  Path: "foo@localhost:8080/user/model"

# After (authority = true)
https://fuel.gazebosim.com/user/model
  Authority: "fuel.gazebosim.com"
  Path: "user/model"

https://foo@localhost:8080/user/model -> 
  Authority: "foo@localhost:8080"
  Path: "user/model"
```

So we need the "full" URL returned by `Url()` to get the Authority as part of the unique name.

Reply:
Ah, I misunderstood `Path` as a function that somehow converts the URL into a file path, not returning the formal `Path`  of a URL. This makes sense. Thanks!